Enhance front end documentation (#41)
BiAPoL/napari-stress: README.md

# napari-stress

> This plugin provides tools for the analysis of surfaces in Napari. It provides utilies to determine and refine the surface-representations of objects with a ray-casting approach and calculate surface curvatures. It re-implements code in Napari that was written for [Gross et al. (2021): STRESS, an automated geometrical characterization of deformable particles for in vivo measurements of cell and tissue mechanical stresses](https://www.biorxiv.org/content/10.1101/2021.03.26.437148v1) and has been made public in [this repository](https://github.com/campaslab/STRESS).

Reviewer: ```suggestion
This plugin provides tools for the analysis of surfaces in Napari, such as utilities to determine and refine the surface-representation of objects using a ray-casting approach and calculate the curvature of surfaces. It re-implements code in Napari that was written for [Gross et al. (2021): STRESS, an automated geometrical characterization of deformable particles for in vivo measurements of cell and tissue mechanical stresses](https://www.biorxiv.org/content/10.1101/2021.03.26.437148v1) and has been made open source in [this repository](https://github.com/campaslab/STRESS).

```

> 

Reviewer 2: The second link to: "[this repository]" is the same link to the paper, I'm not sure that's right

## Usage

Functionality in this repository is divided in two groups: **Recipes** and **plugins**.

### Recipes

> Napari-stress provides jupyter notebooks with [complete workflows](./docs/notebooks/recipes) for different types of input data and objectives. Napari-stress currently provides notebooks for the following data/image types:

Reviewer: I suggest to have napari user-documentation first and code second.

* **Confocal data** (*.tif*), 3D+t: This type of data can be processed with napari-stressed as show in [this notebook](./docs/notebooks/recipes/Process_confocal.ipynb)

![](./docs/imgs/confocal/workflow.png)

> * **Lightsheet data** (*.czi*), 3D + t: coming soon....

Reviewer: If it's coming soon, remove it from the readme and create a github issue instead.

### Plugins

> All functions in this repository are documented sesparately as Jupyter notebooks [here](./docs/notebooks/demo). 

Reviewer: ```suggestion
All functions in this repository are documented separately as Jupyter notebooks [here](./docs/notebooks/demo). 

```

### Utilities

> Data to be used for this plugin is typically of the form `[TZYX]` (e.g., 3D + time). Napari-stress offers some convenient way to use functions from other repositories (which are often made for 3D data) on timelapse data with the `frame_by_frame` function and the `TimelapseConverter` class. Both are described in more detail in [this notebook](https://github.com/BiAPoL/napari-stress/blob/main/docs/notebooks/demo/demo_timelapse_processing.ipynb).

Reviewer: ```suggestion
Data to be used for this plugin is typically of the form `[TZYX]` (e.g., 3D + time). Napari-stress offers convenient ways to use functions from other repositories (which are often made for 3D data) on timelapse data with the `frame_by_frame` function and the `TimelapseConverter` class. Both are described in more detail in [this notebook](https://github.com/BiAPoL/napari-stress/blob/main/docs/notebooks/demo/demo_timelapse_processing.ipynb).

```

> Data to be used for this plugin is typically of the form `[TZYX]` (e.g., 3D + time). Napari-stress offers some convenient way to use functions from other repositories (which are often made for 3D data) on timelapse data with the `frame_by_frame` function and the `TimelapseConverter` class. Both are described in more detail in [this notebook](https://github.com/BiAPoL/napari-stress/blob/add-timelapse-decorator-for-points-and-surfaces/docs/notebooks/demo/TimeLapse_processing.ipynb).

Reviewer 2: The link added here is broken. I get a 404 not found error
